tests/test_application.py

```diff
@@ -1501,6 +1508,51 @@ async def post_shutdown(app: Application) -> None:
             "post_shutdown",
         ], "Wrong order of events detected!"
 
+    @pytest.mark.skipif(
+        platform.system() == "Windows",
+        reason="Can't send signals without stopping whole process on windows",
+    )
+    def test_run_polling_post_stop(self, bot, monkeypatch):
+        events = []
+
+        async def get_updates(*args, **kwargs):
+            # This makes sure that other coroutines have a chance of running as well
+            await asyncio.sleep(0)
+            return []
+
+        def thread_target():
+            waited = 0
+            while not app.running:
+                time.sleep(0.05)
+                waited += 0.05
+                if waited > 5:
+                    pytest.fail("App apparently won't start")
+
+            os.kill(os.getpid(), signal.SIGINT)
+
+        async def post_stop(app: Application) -> None:
+            events.append("post_stop")
+
+        app = Application.builder().token(bot.token).post_stop(post_stop).build()
+        app.bot._unfreeze()
+        monkeypatch.setattr(app.bot, "get_updates", get_updates)
+        monkeypatch.setattr(app, "stop", call_after(app.stop, lambda _: events.append("stop")))
+        monkeypatch.setattr(
+            app.updater,
+            "stop",
+            call_after(app.updater.stop, lambda _: events.append("updater.stop")),
+        )
```

Comment:
can you also monkeypatch `shutdown` to unsure that `post_stop` runs before `shutdown`? :) same below